app/src/main/java/pm/gnosis/heimdall/ui/onboarding/RestoreAccountViewModel.kt

```diff
@@ -0,0 +1,21 @@
+package pm.gnosis.heimdall.ui.onboarding
+
+import io.reactivex.Completable
+import io.reactivex.Single
+import io.reactivex.schedulers.Schedulers
+import pm.gnosis.heimdall.accounts.base.repositories.AccountsRepository
+import javax.inject.Inject
+
+class RestoreAccountViewModel @Inject constructor(private val accountsRepository: AccountsRepository) : RestoreAccountContract() {
+    override fun isValidMnemonic(mnemonic: String): Single<Boolean> {
+        val words = mnemonic.split(Regex("\\s+"))
```

Comment:
Shouldn't that be part of the validateMnemonic function? (I got a bad comment on one of my challenges because this is business logic in the viewmodel class :D )